tests/gtfs/test_validation.py

```diff
@@ -580,3 +602,203 @@ def test_summarise_routes_on_pass(self, gtfs_fixture):
+        save_dir = os.listdir(os.path.join(tmp_path, "save_test"))
+        counts = {"html": 0, "png": 0}
+        for pth in save_dir:
+            if ".html" in pth:
+                counts["html"] += 1
+            elif ".png" in pth:
+                counts["png"] += 1
+
+        assert os.path.exists(
+            os.path.join(tmp_path, "save_test")
+        ), "'save_test' dir could not be created'"
+        assert counts["html"] == 1, "Failed to save plot as HTML"
+        assert counts["png"] == 1, "Failed to save plot as png"
+
+        # save test for an image with invalid file extension
+        valid_img_formats = ["png", "pdf", "jpg", "jpeg", "webp", "svg"]
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Please specify a valid image format. Valid formats "
+                f"include {valid_img_formats}"
+            ),
+        ):
+            gtfs_fixture._plot_summary(
+                gtfs_fixture.daily_route_summary,
+                "route_count_mean",
+                save_image=True,
+                out_dir=os.path.join(tmp_path, "outputs"),
+                img_type="test",
+            )
+
+    def test__plot_route_summary_defences(self, gtfs_fixture):
+        """Test the defences for the small wrapper plot_route_summary()."""
+        # test attribute checks
+        with pytest.raises(
+            AttributeError,
+            match=(
+                re.escape(
+                    "The daily_route_summary table could not be found."
+                    " Did you forget to call '.summarise_routes()' first?"
+                )
+            ),
+        ):
+            gtfs_fixture.plot_route_summary(target_summary="mean")
+
+    def test__plot_trip_summary_defences(self, gtfs_fixture):
+        """Test the defences for the small wrapper plot_trip_summary()."""
+        # test attribute checks
+        with pytest.raises(
+            AttributeError,
+            match=(
+                re.escape(
+                    "The daily_trip_summary table could not be found."
+                    " Did you forget to call '.summarise_trips()' first?"
+                )
+            ),
+        ):
+            gtfs_fixture.plot_trip_summary(target_summary="mean")
+
+    def test__plot_route_summary_on_pass(self, gtfs_fixture):
+        """Test plot_route_summary() calls _plot_summary() succesfully."""
+        gtfs_fixture.summarise_routes()
+        assert isinstance(
+            gtfs_fixture.plot_route_summary(target_summary="mean"),
+            PlotlyFigure,
+        ), "plot_route_summary() failed to return plotly figure"
+
+    def test__plot_trip_summary_on_pass(self, gtfs_fixture):
+        """Test plot_trip_summary() calls _plot_summary() succesfully."""
+        gtfs_fixture.summarise_trips()
+        assert isinstance(
+            gtfs_fixture.plot_trip_summary(target_summary="mean"), PlotlyFigure
+        ), "plot_trip_summary() failed to return plotly figure"
```

Comment:
These tests are clones of each other, testing the same logic in the src code that I suggested would be better refactored to one method.